### PR TITLE
Add Player Inventory

### PR DESCRIPTION
**Pull request by Bryntet**
Hello! As i talked about earlier I've done some work on the inventory system. It's very much still WIP though, but I hope you'll appreciate my additions :)

I'll keep doing things tomorrow, but I'd love to get any input on this if you want to :)

### CODE REVIEW

File: pumpkin-inventory/src/player.rs [hunk 1]
```diff
@@ -2,13 +2,27 @@ use pumpkin_world::item::Item;
 
 pub struct PlayerInventory {
     // Main Inventory + Hotbar
+    crafting: [Option<Item>; 4],
+    crafting_output: Option<Item>,
     items: [Option<Item>; 36],
     armor: [Option<Item>; 4],
     offhand: Option<Item>,
-    // current selected slot in hortbar
-    selected: i16,
```
Comment (Snowiiii) [inline pumpkin-inventory/src/player.rs:9]: Why you changes this to a usize?

File: pumpkin-world/src/item/item_category_checks.rs [hunk 2]
```diff
@@ -0,0 +1,75 @@
+use crate::item::Item;
```
Comment (Snowiiii) [inline pumpkin-world/src/item/item_category_checks.rs:1]: I don't like name of this file. I think item_category would be better

Comment (Bryntet) [reply pumpkin-world/src/item/item_category_checks.rs:1]: maybe `item_categories`?

**Comment by Snowiiii** [comment]
Okay, Looks fine. then. I think it may be better to merge when this get to an more finished state. Maybe we should get block placement from the selected item in the hotbar working ?.

**Comment by Bryntet** [comment]
Agreed. I am actually working on the block placement right now locally. I will update here when done :)

**Comment by Snowiiii** [comment]
> Agreed. I am actually working on the block placement right now locally. I will update here when done :)

Oh Great :+1:

**Comment by Snowiiii** [comment]
Everything looks good. Just fix the conflicts and i think its ready to merge

**Pull request merged**